src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -188,10 +190,32 @@ private string GetRootPathWithGuid_NoLock()
             var fullName = GetFullReflectionName(containingOrThis);
 
             var compilation = await temporaryDocument.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-            // TODO: retrieve path to actual assembly instead of reference assembly
-            var reference = compilation.GetMetadataReference(symbol.ContainingAssembly);
+
+            string assemblyLocation = null;
+            var isReferenceAssembly = symbol.ContainingAssembly.GetAttributes().Any(attribute => attribute.AttributeClass.Name == nameof(ReferenceAssemblyAttribute));
+            if (isReferenceAssembly)
+            {
+                // Attempt to resolve the assembly via the current process' binding path
```

Comment:
I don't think we should be resolving or loading assemblies in devenv process.

Reply:
Is there something in .Net, the project system, VS, dtar (etc. etc.) that would be able to do this for us?

Reply:
No.